### DISCUSSION TITLE
GLIBC version not found when running RPI build

### DISCUSSION THREAD

**Discussion by BillYJT** [nasa/fprime#1922] [Q&A] [answered]
Hi.

I encountered the following error when running the demo RPI build on the raspberry pi.

./RPI: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.33' not found (required by ./RPI)
./RPI: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.32' not found (required by ./RPI)
./RPI: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.34' not found (required by ./RPI)

May I ask if this is a known issue? Is this due to the pi environment lacking the proper GLIBC or incorrect cross-compiling?

Thanks!

**Comment by LeStarch** [comment, marked as answer]
This happens when compiling with a newer cross-compiler than is supported by the operating system on the pi.

In order to get repeatable builds with the PI, we switched our CI to use: https://releases.linaro.org/components/toolchain/binaries/latest-7/arm-linux-gnueabihf/gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz

**Reply by BillYJT** [reply]
That worked. Thanks!

**Reply by jdhttp** [reply]
Hi LeStarch,
I installed Pi Zero 2W 64-bit OS by using your linked image-1.8.5. Windows rdp get black screen although xrdp has been installed in the Pi. Plus  dll --version shows 2.3.1 which errors out GLIBC 2.3.3, GLIBC 2.3.4 not found, while running a compiled exe file. 
Please help for this.
Thank you.
John

**Reply by LeStarch** [reply]
I have heard that the PI zero has other issues with cross-compiling.  @mohitsingh999 once recommended users download a toolchain made specifically for the PI Zero.  A quick internet search should reveal several maintainers of such toolchains.  Alternatively, you could build your own toolchain....but that is outside the scope of my expertise.

I cannot offer much insight into xrdp because I have not used it before.

**Reply by jdhttp** [reply]
Hi LeStarch,
Appreciate your quick response. You're right: both cross-compiling and compiling in Pi Zero 2W (ldd 2.3.1) show "build completed successful", however running these compiled exe enounters error of GLIBC 2.3.3, GLIBC 2.3.4 not found. Please enlighten more.
John
